<a href="https://colab.research.google.com/github/Maatpz/Analise-de-Dados/blob/main/Exercicio%20paratico%207/atividade_pratica_aula4_kpis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Atividade Prática — Aula 4: Análise de Dados e Construção de Indicadores (KPIs)

Esta atividade foi construída com base na Aula 4, que marca a transição **do dado limpo para a tomada de decisão estratégica**, com foco em **KPIs, métricas derivadas, dimensões de análise, rankings e insights executivos**. fileciteturn5file0

## Ideia central da aula
Antes do código, vem a **pergunta de negócio**.  
Depois, definimos:
1. a **métrica**
2. a **dimensão**
3. a **agregação**
4. e só então construímos a resposta em tabela e em texto. fileciteturn5file0

## Regras da atividade
- O notebook **orienta**, mas você deve **escrever os códigos**.
- Sempre que gerar uma tabela agregada, escreva **1 ou 2 frases interpretando o resultado**.
- O objetivo não é apenas calcular números, mas comunicar decisões.

## Dataset da atividade
Arquivo: `vendas_brasil_aula4_kpis.csv`


## 1. Preparação do ambiente

Importe as bibliotecas necessárias para a atividade.

**Sugestão:**  
- `pandas`
- `numpy`

Se desejar, você também pode usar uma biblioteca de visualização mais adiante, mas o foco desta aula é **tabela agregada + insight**.


In [7]:
# Escreva aqui suas importações
import pandas as pd
import numpy as np



## 2. Leitura da base

Leia o arquivo `vendas_brasil_aula4_kpis.csv` em um DataFrame chamado `df`.

Depois:
1. exiba as primeiras linhas
2. veja o tamanho da base
3. inspecione os tipos das colunas
4. confirme se `data_venda` está pronta para análises temporais


In [18]:
# Leia o CSV e faça a inspeção inicial

df = pd.read_csv('/content/Dados/vendas_brasil_aula4_kpis.csv')

# Visualização inicial
df.head()

# Tamanho da base
df.shape

# Tipos de dados
df.dtypes

# Converter data
df['data_venda'] = pd.to_datetime(df['data_venda'])


,pedido_id,data_venda,uf,canal_venda,categoria,produto,quantidade_pedidos,preco_unitario,receita,lucro
0,5000,2025-02-14,RJ,Marketplace,Periféricos,Mouse Gamer,2,237.65,475.30,239.08
1,5001,2025-02-17,PR,Online,Informática,Monitor 27,4,1431.79,5727.16,1769.48
2,5002,2025-07-22,RJ,Marketplace,Informática,Monitor 27,5,1446.22,7231.10,2449.85
3,5003,2025-12-08,SP,Marketplace,Periféricos,Teclado Mecânico,7,399.84,2798.88,1356.67
4,5004,2025-01-04,GO,Marketplace,Telefonia,Smartphone X,4,3134.85,12539.40,3656.84


## 3. Perguntas antes do código

A aula enfatiza que a maior armadilha é começar com `groupby()` sem saber o que se quer responder. fileciteturn5file0

### Tarefa
Antes de programar, escreva em markdown respostas curtas para estas perguntas:

1. Qual canal de venda é o mais eficiente?
Margem de lucro alta
2. Qual categoria parece mais rentável?
a com maior lucro total ou margem
3. Quais produtos são os campeões por lucro?
os com maior lucro
4. O negócio está crescendo ao longo do tempo?
analisando receita/lucro ao longo do tempo
5. Quais KPIs globais resumem melhor a operação?
receita, lucro,margem e ticket médio

Depois disso, avance para o código.


## 4. KPIs globais da operação

Com base nas “4 Métricas de Ouro do Varejo” apresentadas na aula, calcule os seguintes KPIs globais da base inteira: fileciteturn5file0

- **Receita Total**
- **Lucro Total**
- **Margem de Lucro (%)**
- **Ticket Médio**

### Orientação
- Pense primeiro na fórmula de cada KPI
- Escreva os cálculos em células separadas ou em uma pequena tabela-resumo
- Se quiser, formate os resultados para facilitar a leitura


In [19]:
# Calcule aqui os 4 KPIs globais

# 1. Receita Total
receita_total = df['receita'].sum()

# 2. Lucro Total
lucro_total = df['lucro'].sum()

# 3. Margem de Lucro (%)
margem_lucro_percentual = (lucro_total / receita_total) * 100

# 4. Ticket Médio
total_pedidos = df['pedido_id'].nunique() # Número de pedidos únicos
ticket_medio = receita_total / total_pedidos

# Exibir os resultados formatados
print(f"KPIs Globais da Operação:\n")
print(f"Receita Total: R$ {receita_total:,.2f}")
print(f"Lucro Total:   R$ {lucro_total:,.2f}")
print(f"Margem de Lucro: {margem_lucro_percentual:.2f}%")
print(f"Ticket Médio:  R$ {ticket_medio:,.2f}")

KPIs Globais da Operação:

Receita Total: R$ 1,921,492.22
Lucro Total:   R$ 544,496.15
Margem de Lucro: 28.34%
Ticket Médio:  R$ 4,574.98


### Interpretação
Escreva 3 a 5 linhas respondendo:
- O negócio parece grande ou pequeno?
Para relevante sabendo que a receita e alta
- O lucro acompanha bem a receita?
Sim, 28% e uma margem boa
- O ticket médio sugere compras de maior valor ou de menor valor?
Pelo valor do ticket indica compras de maior valor


## 5. Métricas derivadas

A aula mostra que **somar valores não basta**.  
Precisamos criar métricas derivadas para avaliar eficiência e comportamento. fileciteturn5file0

### Tarefa
Crie, quando fizer sentido:
- uma coluna de **margem_lucro** (`lucro / receita`)
- uma lógica para **ticket_medio** da operação ou por grupo

### Reflexão
Explique:
1. Por que receita alta não significa automaticamente melhor desempenho?
2. Em que situação a margem ajuda mais do que a receita bruta?


In [24]:
# Crie aqui as métricas derivadas
df['margem_lucro'] = df['lucro'] / df['receita']
display(df.head())

print("Receita alta não significa desempenho → pode ter custos altos")
print("Margem é melhor quando queremos medir eficiência e rentabilidade")

,pedido_id,data_venda,uf,canal_venda,categoria,produto,quantidade_pedidos,preco_unitario,receita,lucro,margem_lucro
0,5000,2025-02-14,RJ,Marketplace,Periféricos,Mouse Gamer,2,237.65,475.30,239.08,0.503009
1,5001,2025-02-17,PR,Online,Informática,Monitor 27,4,1431.79,5727.16,1769.48,0.308963
2,5002,2025-07-22,RJ,Marketplace,Informática,Monitor 27,5,1446.22,7231.10,2449.85,0.338794
3,5003,2025-12-08,SP,Marketplace,Periféricos,Teclado Mecânico,7,399.84,2798.88,1356.67,0.484719
4,5004,2025-01-04,GO,Marketplace,Telefonia,Smartphone X,4,3134.85,12539.40,3656.84,0.291628


Receita alta não significa desempenho → pode ter custos altos
Margem é melhor quando queremos medir eficiência e rentabilidade


## 6. Fatiando a realidade: análise por dimensão

A aula mostra que um KPI isolado não decide nada.  
É preciso perguntar: **como isso varia por canal, categoria, região e tempo?** fileciteturn5file0

### Tarefa
Liste, em markdown, quais colunas do dataset representam:
- métrica
- dimensão
- tempo

Métricas: receita, lucro,quantidade_pedidos
Dimensões: canal_venda, categoria, uf, produto
Tempo: data_venda


## 7. Análise por canal

### Pergunta de negócio
**Qual canal de venda é o mais eficiente?**

### Tarefa
Crie uma tabela agregada por `canal_venda` contendo, no mínimo:
- Receita Total
- Lucro Total
- Margem Média ou Margem Calculada
- Quantidade de pedidos
- Ticket Médio

### Depois responda
1. Qual canal gera mais receita?

2. Qual canal gera maior eficiência?

3. Se você fosse gestor, priorizaria volume ou margem?




In [27]:
# Gere aqui a tabela agregada por canal_venda

análise_por_canal = df.groupby('canal_venda').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

análise_por_canal['margem_lucro_percentual'] = (análise_por_canal['lucro_total'] / análise_por_canal['receita_total']) * 100
análise_por_canal['ticket_medio'] = análise_por_canal['receita_total'] / análise_por_canal['quantidade_pedidos']

display(análise_por_canal.sort_values(by='receita_total', ascending=False))


,canal_venda,receita_total,lucro_total,quantidade_pedidos,margem_lucro_percentual,ticket_medio
1,Marketplace,657802.09,185802.57,141,28.245968,4665.263050
0,Loja Física,640797.06,174583.16,143,27.244688,4481.098322
2,Online,622893.07,184110.42,136,29.557307,4580.096103


### Insight obrigatório
Escreva 1 ou 2 frases interpretando sua tabela, no estilo executivo:
- não apenas “o Online foi maior”
- mas o que isso **significa para a decisão**

### Respostas às Perguntas

1.  **Qual canal gera mais receita?**
    O canal **Marketplace** gera a maior receita total, o que o torna crucial para o volume de vendas da empresa.

2.  **Qual canal gera maior eficiência?**
    Embora o Marketplace lidere em receita, o canal **Online** apresenta uma margem de lucro percentual superior (29.56% vs 28.25%). Isso indica que o Online é mais eficiente na conversão de receita em lucro.

3.  **Se você fosse gestor, priorizaria volume ou margem?**
    Como gestor, a decisão entre volume e margem dependeria da estratégia atual da empresa. Se o objetivo é ganhar **market share** e maximizar a visibilidade, priorizar o volume via Marketplace pode ser estratégico. No entanto, para **sustentabilidade e rentabilidade** a longo prazo, seria fundamental otimizar o canal Online e investigar oportunidades para melhorar a margem do Marketplace, ou balancear os esforços para capitalizar a eficiência do Online enquanto mantém o volume do Marketplace.

## 8. Análise por categoria

### Pergunta de negócio
**Qual categoria é mais lucrativa e qual parece mais eficiente?**

### Tarefa
Crie uma tabela agregada por `categoria` com:
- Receita Total
- Lucro Total
- Margem
- Ticket Médio

### Depois responda
1. A categoria de maior receita também é a de maior margem?
2. Existe alguma categoria que parece vender bem, mas ser pouco eficiente?


In [29]:
# Gere aqui a tabela agregada por categoria
análise_por_categoria = df.groupby('categoria').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

análise_por_categoria['margem_lucro_percentual'] = (análise_por_categoria['lucro_total'] / análise_por_categoria['receita_total']) * 100
análise_por_categoria['ticket_medio'] = análise_por_categoria['receita_total'] / análise_por_categoria['quantidade_pedidos']

display(análise_por_categoria.sort_values(by='receita_total', ascending=False))

,categoria,receita_total,lucro_total,quantidade_pedidos,margem_lucro_percentual,ticket_medio
2,Telefonia,866292.45,219975.58,109,25.392762,7947.637156
0,Informática,805376.44,219090.15,93,27.203447,8659.961720
3,Áudio,132900.25,55006.02,111,41.388951,1197.299550
1,Periféricos,116923.08,50424.40,107,43.126130,1092.739065


### Respostas às Perguntas

1.  **A categoria de maior receita também é a de maior margem?**
    Não. A categoria de **maior receita** é 'Telefonia' (R$ 866.292,45), mas a categoria de **maior margem de lucro percentual** é 'Periféricos' (43.13%).

2.  **Existe alguma categoria que parece vender bem, mas ser pouco eficiente?**
    A categoria **Informática** apresenta a segunda maior receita (R$ 805.376,44), mas sua margem de lucro (27.20%) é a mais baixa entre as categorias. Isso sugere que, apesar de um alto volume de vendas, a eficiência em transformar receita em lucro é menor, indicando que pode ser uma categoria que vende bem, mas é menos eficiente em termos de rentabilidade.

## 9. Análise por região (UF)

### Pergunta de negócio
**Quais UFs concentram a operação?**

### Tarefa
Monte uma tabela por `uf` com:
- Receita Total
- Lucro Total
- Participação percentual da receita
- Ticket Médio

### Depois responda
1. Quais UFs concentram maior receita?
2. Existe concentração regional?
3. Se fosse necessário expandir ou reforçar logística, por onde você começaria?


In [30]:
# Gere aqui a análise por UF
análise_por_uf = df.groupby('uf').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique')
).reset_index()

# Calcular a participação percentual da receita de cada UF
receita_total_geral = análise_por_uf['receita_total'].sum()
análise_por_uf['participacao_receita_percentual'] = (análise_por_uf['receita_total'] / receita_total_geral) * 100

# Calcular Ticket Médio
análise_por_uf['ticket_medio'] = análise_por_uf['receita_total'] / análise_por_uf['quantidade_pedidos']

# Calcular Margem de Lucro Percentual
análise_por_uf['margem_lucro_percentual'] = (análise_por_uf['lucro_total'] / análise_por_uf['receita_total']) * 100

display(análise_por_uf.sort_values(by='receita_total', ascending=False))

,uf,receita_total,lucro_total,quantidade_pedidos,participacao_receita_percentual,ticket_medio,margem_lucro_percentual
9,SP,246835.50,71022.16,48,12.846032,5142.406250,28.773074
6,RJ,218631.91,63960.39,38,11.378236,5753.471316,29.254828
4,PE,208457.44,58183.95,55,10.848727,3790.135273,27.911669
3,MG,207614.69,61122.25,44,10.804868,4718.515682,29.440234
8,SC,198143.47,60272.32,40,10.311958,4953.586750,30.418525
7,RS,187048.31,49795.96,43,9.734534,4349.960698,26.621978
1,ES,181113.70,49838.07,39,9.425680,4643.941026,27.517559
5,PR,174136.19,46155.64,35,9.062550,4975.319714,26.505484
0,BA,157082.99,42117.30,42,8.175052,3740.071190,26.812133
2,GO,142428.02,42028.11,36,7.412365,3956.333889,29.508316


### Respostas às Perguntas

1.  **Quais UFs concentram maior receita?**
    As UFs que concentram as maiores receitas são **SP (12.85%)**, **RJ (11.38%)**, **PE (10.85%)** e **MG (10.80%)**. Juntas, essas quatro UFs representam uma parcela significativa da receita total da operação.

2.  **Existe concentração regional?**
    Sim, existe uma concentração regional notável. Observamos que as UFs do Sudeste (SP, RJ, MG) e Nordeste (PE) lideram em receita, indicando que a maior parte da operação está concentrada nessas regiões. As demais UFs têm participações menores e mais distribuídas.

3.  **Se fosse necessário expandir ou reforçar logística, por onde você começaria?**
    Com base na concentração de receita, a expansão ou reforço logístico deveria começar pelas UFs com maior participação, como **São Paulo (SP)** e **Rio de Janeiro (RJ)**, para otimizar o atendimento onde o volume é maior. Além disso, seria prudente analisar **Pernambuco (PE)** e **Minas Gerais (MG)** para garantir que a infraestrutura nessas regiões suporte o volume atual e potencial crescimento, considerando sua relevância em receita.

## 10. O motor da análise: agregações (groupby)

Nos slides, a lógica central é:
- `receita -> sum`
- `margem -> mean` ou cálculo derivado
- `produto -> count`
fileciteturn5file0

### Tarefa
Escolha **duas dimensões diferentes** e construa tabelas agregadas mostrando que você entendeu essa lógica.

Exemplos:
- por canal
- por categoria
- por UF
- por mês


In [33]:
# Demonstre aqui o uso de groupby com diferentes agregações

# Garante que a coluna 'data_venda' está no formato datetime
df['data_venda'] = pd.to_datetime(df['data_venda'])

# Agregação por canal_venda
print("\n--- Análise por Canal de Venda ---")
análise_canal_agg = df.groupby('canal_venda').agg(
    receita_total=('receita', 'sum'),
    margem_lucro_media=('margem_lucro', 'mean'),
    total_pedidos=('pedido_id', 'nunique')
).reset_index()
display(análise_canal_agg.sort_values(by='receita_total', ascending=False))

# Agregação por mês
print("\n--- Análise Mensal ---")
df['mes'] = df['data_venda'].dt.month # Extrai o mês da data
análise_mes_agg = df.groupby('mes').agg(
    receita_total=('receita', 'sum'),
    margem_lucro_media=('margem_lucro', 'mean'),
    total_pedidos=('pedido_id', 'nunique')
).reset_index()
display(análise_mes_agg.sort_values(by='receita_total', ascending=False))


--- Análise por Canal de Venda ---


,canal_venda,receita_total,margem_lucro_media,total_pedidos
1,Marketplace,657802.09,0.347194,141
0,Loja Física,640797.06,0.344944,143
2,Online,622893.07,0.352601,136



--- Análise Mensal ---


,mes,receita_total,margem_lucro_media,total_pedidos
10,11,325537.31,0.334427,44
11,12,223512.51,0.378996,46
6,7,196651.96,0.342844,33
0,1,161571.89,0.350445,38
7,8,160240.62,0.342685,37
9,10,158226.19,0.332739,34
3,4,148930.46,0.343454,43
5,6,134685.83,0.350999,33
8,9,109291.03,0.359818,33
1,2,108587.03,0.354747,31


## 11. Campeões e vilões (Top N)

A aula destaca o poder dos rankings curtos e focados. fileciteturn5file0

### Tarefa
Crie:
- Top 10 produtos por **lucro**
- Top 10 produtos por **receita**
- 5 categorias ou UFs com **pior margem** ou pior desempenho

### Perguntas
1. Quem são os “campeões”?
2. Quem são os “vilões”?
3. Que ação tática poderia ser tomada com base nisso?


In [34]:
# Monte aqui os rankings Top N

# Top 10 produtos por lucro
top_10_produtos_lucro = df.groupby('produto').agg(
    lucro_total=('lucro', 'sum')
).reset_index().sort_values(by='lucro_total', ascending=False).head(10)
print("\n--- Top 10 Produtos por Lucro ---")
display(top_10_produtos_lucro)

# Top 10 produtos por receita
top_10_produtos_receita = df.groupby('produto').agg(
    receita_total=('receita', 'sum')
).reset_index().sort_values(by='receita_total', ascending=False).head(10)
print("\n--- Top 10 Produtos por Receita ---")
display(top_10_produtos_receita)

# 5 Categorias com pior margem
piores_5_categorias_margem = df.groupby('categoria').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum')
).reset_index()
piores_5_categorias_margem['margem_lucro_percentual'] = (piores_5_categorias_margem['lucro_total'] / piores_5_categorias_margem['receita_total']) * 100
piores_5_categorias_margem = piores_5_categorias_margem.sort_values(by='margem_lucro_percentual', ascending=True).head(5)
print("\n--- 5 Categorias com Pior Margem de Lucro ---")
display(piores_5_categorias_margem)

# 5 UFs com pior margem
piores_5_uf_margem = df.groupby('uf').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum')
).reset_index()
piores_5_uf_margem['margem_lucro_percentual'] = (piores_5_uf_margem['lucro_total'] / piores_5_uf_margem['receita_total']) * 100
piores_5_uf_margem = piores_5_uf_margem.sort_values(by='margem_lucro_percentual', ascending=True).head(5)
print("\n--- 5 UFs com Pior Margem de Lucro ---")
display(piores_5_uf_margem)


--- Top 10 Produtos por Lucro ---


,produto,lucro_total
4,Notebook Pro,153094.84
5,Smartphone X,118198.25
6,Tablet Plus,101777.33
2,Monitor 27,65995.31
7,Teclado Mecânico,33939.07
0,Caixa de Som,28316.37
1,Headset Pro,26689.65
3,Mouse Gamer,16485.33



--- Top 10 Produtos por Receita ---


,produto,receita_total
4,Notebook Pro,594132.74
5,Smartphone X,489572.08
6,Tablet Plus,376720.37
2,Monitor 27,211243.70
7,Teclado Mecânico,79973.75
0,Caixa de Som,68956.76
1,Headset Pro,63943.49
3,Mouse Gamer,36949.33



--- 5 Categorias com Pior Margem de Lucro ---


,categoria,receita_total,lucro_total,margem_lucro_percentual
2,Telefonia,866292.45,219975.58,25.392762
0,Informática,805376.44,219090.15,27.203447
3,Áudio,132900.25,55006.02,41.388951
1,Periféricos,116923.08,50424.40,43.126130



--- 5 UFs com Pior Margem de Lucro ---


,uf,receita_total,lucro_total,margem_lucro_percentual
5,PR,174136.19,46155.64,26.505484
7,RS,187048.31,49795.96,26.621978
0,BA,157082.99,42117.30,26.812133
1,ES,181113.70,49838.07,27.517559
4,PE,208457.44,58183.95,27.911669


## 12. Análise temporal

A aula reforça que sem tempo a análise fica estática. fileciteturn5file0

### Tarefa
1. Converta `data_venda` para data, se necessário
2. Crie colunas auxiliares como:
   - mês
   - trimestre (opcional)
3. Gere uma tabela agregada por mês com:
   - receita
   - lucro
   - ticket médio ou margem

### Depois responda
1. O lucro parece crescer ao longo do tempo?
2. O último trimestre foi melhor que o anterior?
3. Quais meses concentram as vendas de fim de ano?


In [35]:
# Faça aqui a análise temporal mensal

# 1. Converta data_venda para data, se necessário
df['data_venda'] = pd.to_datetime(df['data_venda'])

# 2. Crie colunas auxiliares
df['mes'] = df['data_venda'].dt.month
df['trimestre'] = df['data_venda'].dt.quarter

# 3. Gere uma tabela agregada por mês
análise_temporal_mensal = df.groupby('mes').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    quantidade_pedidos=('pedido_id', 'nunique') # Para calcular o ticket médio
).reset_index()

análise_temporal_mensal['ticket_medio'] = análise_temporal_mensal['receita_total'] / análise_temporal_mensal['quantidade_pedidos']
análise_temporal_mensal['margem_lucro_percentual'] = (análise_temporal_mensal['lucro_total'] / análise_temporal_mensal['receita_total']) * 100

display(análise_temporal_mensal.sort_values(by='mes'))

,mes,receita_total,lucro_total,quantidade_pedidos,ticket_medio,margem_lucro_percentual
0,1,161571.89,46515.14,38,4251.891842,28.789129
1,2,108587.03,29998.87,31,3502.807419,27.626568
2,3,87902.22,23923.66,25,3516.088800,27.216218
3,4,148930.46,43094.00,43,3463.499070,28.935652
4,5,106355.17,30334.32,23,4624.137826,28.521716
5,6,134685.83,38592.70,33,4081.388788,28.653868
6,7,196651.96,57913.44,33,5959.150303,29.449714
7,8,160240.62,45881.36,37,4330.827568,28.632790
8,9,109291.03,31938.44,33,3311.849394,29.223295
9,10,158226.19,41463.02,34,4653.711471,26.204903


## 13. Fluxo de investigação do analista

A aula apresenta a sequência:
**Pergunta -> Colunas necessárias -> Tipo de resposta/gráfico**.

### Tarefa
Preencha, em markdown, pelo menos 3 exemplos no formato:

- **Pergunta de negócio:** Qual o desempenho de vendas do produto 'Smartphone X' nos últimos 6 meses?
- **Colunas necessárias:** `data_venda`, `produto`, `receita`, `lucro`
- **Métrica/KPI:** Receita Total, Lucro Total, Margem de Lucro
- **Tipo de resposta esperada:** Gráfico de linha (tendência mensal), tabela de resumo

- **Pergunta de negócio:** Quais são as categorias de produto que mais contribuem para o lucro no canal 'Online'?
- **Colunas necessárias:** `canal_venda`, `categoria`, `lucro`
- **Métrica/KPI:** Lucro Total por Categoria
- **Tipo de resposta esperada:** Ranking (Top N) em tabela

- **Pergunta de negócio:** Existe alguma sazonalidade nas vendas em termos de receita e lucro ao longo do ano?
- **Colunas necessárias:** `data_venda`, `receita`, `lucro`
- **Métrica/KPI:** Receita Total Mensal, Lucro Total Mensal
- **Tipo de resposta esperada:** Gráfico de linha (tendência mensal), tabela agregada por mês

## 14. O insight exige palavras

Segundo a aula, o mercado valoriza quem comunica decisões, não apenas quem roda código. fileciteturn5file0

### Tarefa
Escolha **duas tabelas agregadas** que você criou e escreva, para cada uma, **1 ou 2 frases de interpretação executiva**.

Exemplo de estilo:
- “O canal X concentra maior volume de receita, mas o canal Y apresenta margem superior.”
- “Recomenda-se revisar custos/expansão/priorização com base nesse resultado.”

### Interpretação das tabelas agregadas

**Análise por Canal de Venda:**
O canal Marketplace concentra o maior volume de receita, o que o torna fundamental para o faturamento total. Contudo, o canal Online apresenta uma margem de lucro superior, indicando maior eficiência na conversão de vendas em lucro e sugerindo um potencial de rentabilidade a ser explorado.

**Análise por Categoria:**
A categoria de Telefonia gera a maior receita, impulsionando o volume de vendas da empresa. Por outro lado, a categoria Periféricos demonstra uma margem de lucro significativamente mais alta, o que indica que, embora possa ter um volume menor, é mais rentável por venda e pode ser uma oportunidade para revisar custos ou estratégias de precificação em outras categorias.

## 15. Case Varejo Brasil — Entrega final

A aula propõe quatro entregas centrais para o gestor. fileciteturn5file0

### Sua missão final
Organize o notebook para entregar:

1.  **KPIs Globais**
2.  **Tabelas por Dimensões Críticas**
    - por canal
    - por categoria
3.  **Top 10 Produtos por Lucro**
4.  **Conclusão final com pelo menos uma decisão tática**

### Importante
A conclusão final deve ser escrita em linguagem clara, direta e orientada à decisão.

---

## **Entrega Final para o Gestor: Varejo Brasil - Análise de Performance e Decisões Estratégicas**

### 1. KPIs Globais da Operação

A análise global da empresa revela os seguintes indicadores chave:

*   **Receita Total:** R$ 1,921,492.22
*   **Lucro Total:** R$ 544,496.15
*   **Margem de Lucro:** 28.34%
*   **Ticket Médio:** R$ 4,574.98

**Interpretação:** A operação demonstra um volume de receita considerável com uma margem de lucro saudável, indicando uma boa eficiência geral na conversão de vendas em lucro. O ticket médio elevado sugere que as transações tendem a ser de maior valor.

### 2. Tabelas por Dimensões Críticas

**Análise por Canal de Venda:**

| canal_venda | receita_total | lucro_total | quantidade_pedidos | margem_lucro_percentual | ticket_medio |
|:------------|--------------:|------------:|-------------------:|------------------------:|-------------:|
| Loja Física | 640797.06     | 174583.16   | 143                | 27.24                   | 4481.10      |
| Marketplace | 657802.09     | 185802.57   | 141                | 28.25                   | 4665.26      |
| Online      | 622893.07     | 184110.42   | 136                | 29.56                   | 4580.10      |

**Insight Executivo:** O canal Marketplace lidera em receita total, sendo vital para o volume de vendas. No entanto, o canal Online exibe a maior margem de lucro, indicando uma superior eficiência na rentabilidade. Recomenda-se investigar e replicar as práticas de sucesso do Online no Marketplace para otimizar sua margem, mantendo o alto volume.

**Análise por Categoria:**

| categoria   | receita_total | lucro_total | quantidade_pedidos | margem_lucro_percentual | ticket_medio |
|:------------|--------------:|------------:|-------------------:|------------------------:|-------------:|
| Informática | 805376.44     | 219090.15   | 93                 | 27.20                   | 8659.96      |
| Periféricos | 116923.08     | 50424.40    | 107                | 43.13                   | 1092.74      |
| Telefonia   | 866292.45     | 219975.58   | 109                | 25.39                   | 7947.64      |
| Áudio       | 132900.25     | 55006.02    | 111                | 41.39                   | 1197.30      |

**Insight Executivo:** A categoria Telefonia é a maior geradora de receita, mas a margem de lucro dos Periféricos é significativamente superior. É crucial analisar a estrutura de custos e precificação da Telefonia e Informática para melhorar sua rentabilidade, enquanto se explora o potencial de crescimento dos Periféricos e Áudio, que são mais eficientes.

### 3. Top 10 Produtos por Lucro

| produto          | lucro_total |
|:-----------------|------------:|
| Notebook Pro     | 153094.84   |
| Smartphone X     | 118198.25   |
| Tablet Plus      | 101777.33   |
| Monitor 27       | 65995.31    |
| Teclado Mecânico | 33939.07    |
| Caixa de Som     | 28316.37    |
| Headset Pro      | 26689.65    |
| Mouse Gamer      | 16485.33    |

**Insight Executivo:** Produtos como 'Notebook Pro', 'Smartphone X' e 'Tablet Plus' são os principais motores de lucro da empresa. É fundamental garantir a disponibilidade, promover esses itens e analisar suas estratégias de venda para maximizar ainda mais seu desempenho.

### 4. Conclusão Final e Decisão Tática

A análise dos KPIs globais e das dimensões críticas de canal e categoria revela que, embora a operação seja robusta, existem oportunidades claras de otimização da rentabilidade. A decisão tática recomendada é **focar na melhoria da margem de lucro das categorias 'Telefonia' e 'Informática'**. Isso pode ser alcançado através da renegociação com fornecedores para reduzir custos de aquisição, otimização de campanhas de marketing para produtos de maior margem dentro dessas categorias, ou ajuste estratégico de preços. Simultaneamente, deve-se **proteger e expandir a atuação nos canais e categorias de alta margem como 'Online' e 'Periféricos'**, replicando suas melhores práticas. Essa abordagem permitirá não apenas manter o volume de receita, mas também impulsionar a rentabilidade geral do negócio.

## 16. Desafio extra (opcional)

Monte uma pequena tabela executiva final com:
- Receita Total
- Lucro Total
- Margem
- Ticket Médio

e depois crie uma segunda tabela por canal com os mesmos indicadores.

Compare os resultados e explique:
- o KPI global esconde alguma diferença importante entre os canais?


In [ ]:
# Desafio extra opcional


## 17. Entrega esperada

Seu notebook deve demonstrar:
- organização
- fórmulas corretas de KPI
- uso adequado de `groupby`, `agg`, `sort_values`
- leitura temporal coerente
- interpretação escrita dos resultados

### Mensagem principal da aula
Transformar linhas de dados limpos em **apoio real para decisões de negócio**. fileciteturn5file0


In [36]:
# Desafio extra opcional

# 1. Tabela executiva final com KPIs globais
global_kpis = pd.DataFrame({
    'KPI': ['Receita Total', 'Lucro Total', 'Margem de Lucro (%)', 'Ticket Médio'],
    'Valor': [
        f"R$ {df['receita'].sum():,.2f}",
        f"R$ {df['lucro'].sum():,.2f}",
        f"{(df['lucro'].sum() / df['receita'].sum()) * 100:.2f}%",
        f"R$ {(df['receita'].sum() / df['pedido_id'].nunique()):,.2f}"
    ]
})
print("--- KPIs Globais ---")
display(global_kpis)

# 2. Tabela por canal com os mesmos indicadores
kpis_por_canal = df.groupby('canal_venda').agg(
    receita_total=('receita', 'sum'),
    lucro_total=('lucro', 'sum'),
    total_pedidos=('pedido_id', 'nunique')
).reset_index()

kpis_por_canal['margem_lucro_percentual'] = (kpis_por_canal['lucro_total'] / kpis_por_canal['receita_total']) * 100
kpis_por_canal['ticket_medio'] = kpis_por_canal['receita_total'] / kpis_por_canal['total_pedidos']

# Formatar para exibição
kpis_por_canal['receita_total'] = kpis_por_canal['receita_total'].apply(lambda x: f"R$ {x:,.2f}")
kpis_por_canal['lucro_total'] = kpis_por_canal['lucro_total'].apply(lambda x: f"R$ {x:,.2f}")
kpis_por_canal['margem_lucro_percentual'] = kpis_por_canal['margem_lucro_percentual'].apply(lambda x: f"{x:.2f}%")
kpis_por_canal['ticket_medio'] = kpis_por_canal['ticket_medio'].apply(lambda x: f"R$ {x:,.2f}")

print("\n--- KPIs por Canal de Venda ---")
display(kpis_por_canal.set_index('canal_venda'))


--- KPIs Globais ---


,KPI,Valor
0,Receita Total,"R$ 1,921,492.22"
1,Lucro Total,"R$ 544,496.15"
2,Margem de Lucro (%),28.34%
3,Ticket Médio,"R$ 4,574.98"



--- KPIs por Canal de Venda ---


,receita_total,lucro_total,total_pedidos,margem_lucro_percentual,ticket_medio
canal_venda,,,,,
Loja Física,"R$ 640,797.06","R$ 174,583.16",143,27.24%,"R$ 4,481.10"
Marketplace,"R$ 657,802.09","R$ 185,802.57",141,28.25%,"R$ 4,665.26"
Online,"R$ 622,893.07","R$ 184,110.42",136,29.56%,"R$ 4,580.10"


### Comparação e Interpretação

**O KPI global esconde alguma diferença importante entre os canais?**

Sim, o KPI global de margem de lucro (28.34%) esconde diferenças importantes entre os canais. Embora a margem global seja saudável, ao analisar por canal, percebemos que o **canal Online** possui a maior margem (29.56%), indicando ser o mais eficiente. Em contraste, a **Loja Física** apresenta a menor margem (27.24%).

Essa diferença, mesmo que pareça pequena em termos percentuais, representa uma variação significativa na rentabilidade por venda e na eficiência operacional de cada canal. Um gestor que olhasse apenas para a margem global poderia não identificar a oportunidade de otimizar os canais de menor margem ou de replicar as melhores práticas do canal mais eficiente.